Calculator/ViewController.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Oleksii Kolomiiets on 4/4/18.
+//  Copyright © 2018 Oleksii Kolomiiets. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        inTheMiddleOfTyping = false
+    }
+    
+    @IBOutlet weak var display: UILabel!
+    
+    private var inTheMiddleOfTyping: Bool!
+    var result = 0.0 {
+        didSet {
+            displayValue = result
+        }
+    }
+    var previousValue = 0.0
+    var operation = ""
+    var doingSomeOperation = false
+    var displayValue: Double {
+        get {
+            return Double(display.text!)!
```

Comment:
You can get crash in case display.text would be nil. Could make it more safe?

Reply:
Yes, I can. The first that comes to mind - to check if that  is not a nil and also if that is a digit. Are there more clever ways to do it more safe?

Reply:
if let displayText = display.text {
                return Double(displayText)!
            } else {
                return 0.0
            }